.cbmc-batch/jobs/aws_cryptosdk_md_finish/aws_cryptosdk_md_finish_harness.c

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/cryptosdk/private/cipher.h>
+#include <cbmc_invariants.h>
+#include <evp_utils.h>
+#include <make_common_data_structures.h>
+#include <proof_helpers/proof_allocators.h>
+
+#include <cipher_openssl.h>
+
+void aws_cryptosdk_md_finish_harness() {
+    /* arguments */
+    struct aws_cryptosdk_md_context *md_context = can_fail_malloc(sizeof(struct aws_cryptosdk_md_context));
+    size_t buf_size;
+    void *buf = can_fail_malloc(buf_size);
+    size_t length;
+
+    /* assumptions */
+    __CPROVER_assume(md_context);
```

Comment:
Same as before. If you're going to assume its not null, just stack allocate with a comment.

Reply:
In this case it needs to be dynamically allocated, because the function being verified frees the context.